compiler/src/iree/compiler/Dialect/HAL/Target/LLVMCPU/Builtins/UKernel.cpp

```diff
@@ -27,96 +27,39 @@ loadUKernelBitcodeFile(StringRef filename, llvm::LLVMContext &context) {
       return llvm::parseBitcodeFile(bitcodeBufferRef, context);
     }
   }
+
   // Some bitcode files are optional: we don't have arch-specific ukernel code
   // for all architectures. So it's normal to be returning nullptr here.
   return nullptr;
 }
 
-static void removeTargetAttributes(llvm::Module &module) {
-  // Copied from Device.cpp - TODO: move this to a shared utility.
-  // Clang adds its own per-function attributes that we need to strip so that
-  // our current executable variant target is used instead.
-  for (auto &func : module.functions()) {
-    func.removeFnAttr("target-cpu");
-    func.removeFnAttr("tune-cpu");
-    func.removeFnAttr("target-features");
-  }
-}
-
-llvm::Expected<std::unique_ptr<llvm::Module>>
-loadUKernelBaseBitcode(llvm::TargetMachine *targetMachine,
-                       llvm::LLVMContext &context) {
-  llvm::Triple triple = targetMachine->getTargetTriple();
-  StringRef filename;
-  if (triple.isArch64Bit()) {
-    filename = "ukernel_bitcode_64bit_base.bc";
-  } else if (triple.isArch32Bit()) {
-    filename = "ukernel_bitcode_32bit_base.bc";
-  } else {
-    return llvm::createStringError(
-        llvm::inconvertibleErrorCode(),
-        "Don't know what ukernel bitcode file to load.");
-  }
-  llvm::Expected<std::unique_ptr<llvm::Module>> bitcode =
-      loadUKernelBitcodeFile(filename, context);
-  if (!bitcode) {
-    // Propagate the error to the caller.
-    return bitcode;
-  }
-
-  if (!bitcode.get()) {
-    // File not found. For base bitcode, this shouldn't happen.
-    return llvm::createStringError(llvm::inconvertibleErrorCode(),
-                                   "Base ukernel bitcode file not found: %s",
-                                   filename.str().c_str());
-  }
-
-  // Base bitcode is compiled for any reasonable architecture of the right
-  // bitness, as we don't care about anything else than bitness here.
-  removeTargetAttributes(*bitcode.get());
-  return bitcode;
-}
-
 llvm::Expected<std::unique_ptr<llvm::Module>>
-loadUKernelArchEntryPointsBitcode(llvm::TargetMachine *targetMachine,
-                                  llvm::LLVMContext &context) {
+loadUKernelBitcode(llvm::TargetMachine *targetMachine,
+                   llvm::LLVMContext &context) {
   const char *archName =
       getIreeArchNameForTargetTriple(targetMachine->getTargetTriple());
   char filename[64];
```

Comment:
let's fix this to be C++ while in here - the arbitrary limit here will be tricky to debug if someone comes along with a longer arch string - `std::string("ukernel_bitcode_") + archName + ".bc"`

Reply:
done